src/Raven.Embedded/EmbeddedServer.cs

```diff
@@ -193,13 +229,20 @@ private void ShutdownServerProcess(Process process)
             }
         }
 
-        private async Task<(Uri ServerUrl, Process ServerProcess)> RunServer(ServerOptions options)
+        public event EventHandler? ServerProcessExited;
```

Comment:
We should add `ServerProcessExitedEventArgs` here. They can be empty for now on, but it is easier to expand later on.